From 9a6d2d6b66ee135e5437c2a0b28a6274a4f5a03f Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Sun, 13 Jul 2014 18:00:15 +1000
Subject: [PATCH 1/4] metadata: Properly clean inetref when searching for
 collection

Also reduce duplicated code.
Remove code handling of title as inetref. The issue if it exists must be resolved elsewhere

(cherry picked from commit 7a0fb8ecbfec5a73300df8837ea0ed68877374b9)
---
 mythtv/libs/libmythmetadata/metadatagrabber.cpp |   61 ++++++++---------------
 mythtv/libs/libmythmetadata/metadatagrabber.h   |    1 +
 2 files changed, 21 insertions(+), 41 deletions(-)

diff --git a/mythtv/libs/libmythmetadata/metadatagrabber.cpp b/mythtv/libs/libmythmetadata/metadatagrabber.cpp
index 964e704..bb30593 100644
--- a/mythtv/libs/libmythmetadata/metadatagrabber.cpp
+++ b/mythtv/libs/libmythmetadata/metadatagrabber.cpp
@@ -74,10 +74,6 @@ void InitializeStaticMaps(void)
     }
 }
 
-static QRegExp retagref("^([a-zA-Z0-9_\\-\\.]+\\.[a-zA-Z0-9]{1,3}):(.*)");
-static QMutex reLock;
-
-
 GrabberList MetaGrabberScript::GetList(bool refresh)
 {
     return MetaGrabberScript::GetList(kGrabberAll, refresh);
@@ -263,6 +259,8 @@ MetaGrabberScript MetaGrabberScript::FromTag(const QString &tag,
 MetaGrabberScript MetaGrabberScript::FromInetref(const QString &inetref,
                                                  bool absolute)
 {
+    static QRegExp retagref("^([a-zA-Z0-9_\\-\\.]+\\.[a-zA-Z0-9]{1,3}):(.*)");
+    static QMutex reLock;
     QMutexLocker lock(&reLock);
 
     if (retagref.indexIn(inetref) > -1)
@@ -278,6 +276,19 @@ MetaGrabberScript MetaGrabberScript::FromInetref(const QString &inetref,
     return MetaGrabberScript();
 }
 
+QString MetaGrabberScript::CleanedInetref(const QString &inetref)
+{
+    static QRegExp retagref("^([a-zA-Z0-9_\\-\\.]+\\.[a-zA-Z0-9]{1,3}):(.*)");
+    static QMutex reLock;
+    QMutexLocker lock(&reLock);
+
+    // try to strip grabber tag from inetref
+    if (retagref.indexIn(inetref) > -1)
+        return retagref.cap(2);
+
+    return inetref;
+}
+
 MetaGrabberScript::MetaGrabberScript(void) :
     m_name(""), m_author(""), m_thumbnail(""), m_fullcommand(""), m_command(""),
     m_type(kGrabberInvalid), m_typestring(""), m_description(""), m_accepts(),
@@ -483,17 +494,8 @@ MetadataLookupList MetaGrabberScript::Search(const QString &title,
     QStringList args;
     SetDefaultArgs(args);
 
-    QString tmptitle = title;
-    {
-        // television grabber may search using inetref for some reason, so test
-        // it for a grabber tag
-        QMutexLocker lock(&reLock);
-        if (retagref.indexIn(title) > -1)
-            tmptitle = retagref.cap(2);
-    }
-
     args << "-M"
-         << tmptitle;
+         << title;
 
     return RunGrabber(args, lookup, passseas);
 }
@@ -505,15 +507,8 @@ MetadataLookupList MetaGrabberScript::SearchSubtitle(const QString &title,
     QStringList args;
     SetDefaultArgs(args);
 
-    QString tmptitle = title;
-    {
-        QMutexLocker lock(&reLock);
-        if (retagref.indexIn(title) > -1)
-            tmptitle = retagref.cap(2);
-    }
-
     args << "-N"
-         << tmptitle
+         << title
          << subtitle;
 
     return RunGrabber(args, lookup, passseas);
@@ -525,16 +520,8 @@ MetadataLookupList MetaGrabberScript::LookupData(const QString &inetref,
     QStringList args;
     SetDefaultArgs(args);
 
-    QString tmpref = inetref;
-    {
-        // try to strip grabber tag from inetref
-        QMutexLocker lock(&reLock);
-        if (retagref.indexIn(inetref) > -1)
-            tmpref = retagref.cap(2);
-    }
-
     args << "-D"
-         << tmpref;
+         << CleanedInetref(inetref);
 
     return RunGrabber(args, lookup, passseas);
 }
@@ -546,16 +533,8 @@ MetadataLookupList MetaGrabberScript::LookupData(const QString &inetref,
     QStringList args;
     SetDefaultArgs(args);
 
-    QString tmpref = inetref;
-    {
-        // try to strip grabber tag from inetref
-        QMutexLocker lock(&reLock);
-        if (retagref.indexIn(inetref) > -1)
-            tmpref = retagref.cap(2);
-    }
-
     args << "-D"
-         << tmpref
+         << CleanedInetref(inetref)
          << QString::number(season)
          << QString::number(episode);
 
@@ -570,7 +549,7 @@ MetadataLookupList MetaGrabberScript::LookupCollection(
     SetDefaultArgs(args);
 
     args << "-C"
-         << collectionref;
+         << CleanedInetref(collectionref);
 
     return RunGrabber(args, lookup, passseas);
 }
diff --git a/mythtv/libs/libmythmetadata/metadatagrabber.h b/mythtv/libs/libmythmetadata/metadatagrabber.h
index cb30581..7d64e17 100644
--- a/mythtv/libs/libmythmetadata/metadatagrabber.h
+++ b/mythtv/libs/libmythmetadata/metadatagrabber.h
@@ -50,6 +50,7 @@ class META_PUBLIC MetaGrabberScript : public QObject
                                         bool absolute=false);
     static MetaGrabberScript    FromInetref(const QString &inetref,
                                             bool absolute=false);
+    static QString              CleanedInetref(const QString &inetref);
 
     bool          IsValid(void) const         { return m_valid; }
 
-- 
1.7.10.2

